docs/proposals/templates-v2.md

```diff
@@ -0,0 +1,334 @@
+
+`Foam: Create New Note` will offer the fastest workflow for creating a note when you don't need customization, while `Foam: Create New Note From Template` will remain to serve a fully customizable (but slower) workflow.
+
+#### Case 1: `foam.defaultNoteTemplate === null`
+
+If `foam.defaultNoteTemplate === null`, it behaves the same as `Markdown Notes: New Note`:
+  * it would ask for a note name and create the note in the current directory. It would open a note with the note name as the title.
+
+**Note:** this would use an implicit default template, making use of the `${title}` variable.
+
+#### Case 2: `foam.defaultNoteTemplate` defined
+
+If `foam.defaultNoteTemplate` is defined:
+  * it asks for the note name and creates the note in the current directory
+
+**Progress:** At this point, we have a faster way to create new notes from templates.
+
+### Change missing wikilinks to use the default template
+
+Clicking on a dangling/missing wikilink should be equivalent to calling `Foam: Create New Note` with the contents of the link as the title.
+That way, creating a note by navigating to a missing note uses the default template.
+
+### Add a metadata section to templates
+
+* The `Foam: New Note` command creates a new note in the current directory. This is a sensible default that makes it quick, but lacks flexibility.
+* The `Foam: Create New Note From Template` asks the user to confirm/customize the filepath. This is more flexible but slower since there are more steps involved.
+
+Both commands use templates. It would be nice if we could template the filepaths as well as the template contents (See ["Template the filepath in `openDailyNote`"](https://github.com/foambubble/foam/issues/523) for a more in-depth discussion the benefits of filepath templating).
+
+In order to template the filepath, there needs to be a place where metadata like this can be specified.
+I think this metadata should be stored alongside the templates themselves. That way, it can make use of all the same template variable available to the templates themselves.
+
+Conceptually, adding metadata to the templates is similar to Markdown frontmatter, though the choice of exact syntax for adding this metadata will have to be done with care since the templates can contain arbitrary contents including frontmatter.
+
+#### Example
+
+A workable syntax is still to be determined.
+While this syntax probably doesn't work as a solution, for this example I will demonstrate the concept using a second frontmatter block:
+
+```markdown
+<!-- The below front-matter block is for foam-specific template settings -->
+---
+<!-- The default filepath to use when using this template -->
+<!-- Relative paths are relative to the workspace, absolute paths are absolute -->
+<!-- Note that you can include VSCode snippet variables to template the path -->
+filepath: `journal/${CURRENT_YEAR}-${CURRENT_MONTH}-${CURRENT_DATE}_#{title}.md`
```

Comment:
All names can be changed to whatever makes the most sense.

----

If I had to justify it, I'd say that VSCode system-defined variables use ALL CAPS and it could make sense to distinguish Foam/application-defined variables from those. 🤷 Happy to do whatever makes the most sense.

Reply:
the question was more about `$` vs `#`, but yes the casing as well should just be consistent. If we want we could namespace the Foam variables (e.g. `${FOAM_TITLE}`)?

Reply:
> the question was more about $ vs #

Whoops. That's my Ruby leaking in... Fixed.